api/layer/tree_mock.go

```diff
@@ -390,7 +390,18 @@ func (t *TreeServiceMock) GetLastPart(ctx context.Context, bktInfo *data.BucketI
 		return 1
 	})
 
-	return parts[len(parts)-1], nil
+	var pi *data.PartInfo
+	for _, part := range parts {
```

Comment:
aren't parts sorted?

Reply:
why not try to get part directly via slice index? is there any cases when N-th part is not N-th in slice?

Reply:
There may be more than one part with the same index. They are sorted by `ServerCreated` field. We should find the latest/actual one.

Reply:
But why? Even when you reupload, you invalidate the old one, but you still have only one proper current one.

Reply:
What do you mean invalidate? There is no such functionality. Only [add and get](https://github.com/nspcc-dev/neofs-s3-gw/blob/06acb581f7b68f0b7412e0dd340fb25f75d7aef1/api/layer/tree_service.go#L82). Any reuploaded parts presented in the list.

Reply:
We may try to filter out duplicates on the level down and try to have there is only valid part list

Reply:
OK, seems to be unavoidable now.